packages/blockchain/egld/src/lib/services/egld.offchain.ts

```diff
@@ -0,0 +1,73 @@
+import { ApiServices, Currency, TransferEth, TransferEthKMS, TransferEthMnemonic } from '@tatumio/api-client'
+import { abstractBlockchainOffchain } from '@tatumio/shared-blockchain-abstract'
+import { Blockchain } from '@tatumio/shared-core'
+import BigNumber from 'bignumber.js'
+import { from } from 'form-data'
+import { prepareSignedTransaction } from './egld.tx'
+
+export const egldOffchainService = (args: { blockchain: Blockchain }) => {
+  return {
+    ...abstractBlockchainOffchain(args),
+    send,
+  }
+}
+
+type EgldOffchain = TransferEth | TransferEthMnemonic | TransferEthKMS
+
+/**
+ * Send EGLD transaction from Tatum Ledger account to the blockchain. This method broadcasts signed transaction to the blockchain.
+ * This operation is irreversible.
+ * @param body content of the transaction to broadcast
+ * @returns transaction id of the transaction in the blockchain or id of the withdrawal, if it was not cancelled automatically
+ */
+export const send = async (body: EgldOffchain) => {
+  if ('signatureId' in body) {
+    return await ApiServices.offChain.blockchain.egldTransfer(body as TransferEthKMS)
+  }
+
+  const { gasLimit, gasPrice, ...withdrawal } = body
+  let fromPriv: string
+
+  if ('mnemonic' in body) {
+    fromPriv = (await ApiServices.blockchain.elgo.egldGenerateAddressPrivateKey({
+      mnemonic: body.mnemonic,
+      index: body.index,
+    })) as string
+  } else fromPriv = body.privateKey
+
+  const fee = {
+    gasLimit: `${gasLimit || '50000'}`,
+    gasPrice: `${gasPrice || '1000000000'}`,
+  }
+
+  const txData = await prepareSignedTransaction({
+    amount: body.amount,
+    fromPrivateKey: fromPriv,
+    fee,
+    to: body.address,
+  })
+
+  const { id } = await ApiServices.offChain.withdrawal.storeWithdrawal({
+    ...withdrawal,
+    fee: new BigNumber(fee.gasLimit).multipliedBy(fee.gasPrice).toString(),
+  })
+
+  try {
```

Comment:
I'm thinking about this try/catch if there is any way how to make it more simple. I would:

- Think if console is needed.
- Add return type to the function - probably union of `broadcastBlockchainTransaction` and `cancelInProgressWithdrawal`
- In case of 
```
    } catch (e1) {
       console.log(e)
       return { id }
     }
```
I would directly throw the error.

Reply:
I think here the id needs to be returned because with this id you can cancel withdrawal manually if transaction was not valid in blockchain
I will add return types and remove console.log

Reply:
I mean if there will be some error while canceling in the previous block that you can still cancel it manually with the id that will be returned

Reply:
Hmm, makes sense.